### PR TITLE
[SPARK-12548][Build] Add  more exceptions to Guava relocation

### PR DESCRIPTION
**Pull request by microhello**
Add `com.google.common.base.AbstractIterator` exceptions to Guava relocation , So far, all guava 14.0.1 `Optional` Class dependencies have added;

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,14 +99,14 @@
...
     <module>assembly</module>
-    <module>external/twitter</module>
-    <module>external/flume</module>
-    <module>external/flume-sink</module>
-    <module>external/flume-assembly</module>
-    <module>external/mqtt</module>
-    <module>external/mqtt-assembly</module>
-    <module>external/zeromq</module>
-    <module>examples</module>
+    <!--<module>external/twitter</module>-->
```
Comment (srowen) [inline pom.xml]: Most of these changes you've made can't be merged. Do you mind closing this PR?

**Comment by srowen** [comment]
Why does this need to be added? 
The Guava exceptions are going away in 2.x anyway.
This also doesn't follow https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark

File: pom.xml (hunk 1 above)
Comment (microhello) [inline pom.xml]: These changes only used by myself, don't need  merge. But I hope the following changes be merged:
Add `com.google.common.base.AbstractIterator` exceptions to Guava relocation , So far, all guava 14.0.1 `Optional` Class dependencies have added;

**Comment by microhello** [comment]
@srowen 
When I use  [facebook/swift](https://github.com/facebook/swift)  with spark 1.5.2, throw following exception, so I need add `com.google.common.base.AbstractIterator` exceptions to Guava relocation . 
```java
Exception in thread "main" java.lang.IllegalAccessError: class com.google.common.base.Optional$1$1 cannot access its superclass org.spark-project.guava.base.AbstractIterator
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.google.common.base.Optional$1.iterator(Optional.java:223)
	at com.google.common.collect.ImmutableSet.copyOf(ImmutableSet.java:300)
	at com.facebook.swift.codec.metadata.AbstractThriftMetadataBuilder.inferThriftFieldIds(AbstractThriftMetadataBuilder.java:566)
	at com.facebook.swift.codec.metadata.AbstractThriftMetadataBuilder.inferThriftFieldIds(AbstractThriftMetadataBuilder.java:542)
	at com.facebook.swift.codec.metadata.AbstractThriftMetadataBuilder.normalizeThriftFields(AbstractThriftMetadataBuilder.java:483)
	at com.facebook.swift.codec.metadata.ThriftStructMetadataBuilder.<init>(ThriftStructMetadataBuilder.java:48)
	at com.facebook.swift.codec.metadata.ThriftCatalog.extractThriftStructMetadata(ThriftCatalog.java:542)
	at com.facebook.swift.codec.metadata.ThriftCatalog.getThriftStructMetadata(ThriftCatalog.java:408)
	at com.facebook.swift.codec.metadata.ThriftCatalog.getThriftTypeUncached(ThriftCatalog.java:283)
	at com.facebook.swift.codec.metadata.ThriftCatalog.getThriftType(ThriftCatalog.java:215)
	at com.facebook.swift.service.metadata.ThriftMethodMetadata.<init>(ThriftMethodMetadata.java:90)
	at com.facebook.swift.service.metadata.ThriftServiceMetadata.<init>(ThriftServiceMetadata.java:73)
	at com.facebook.swift.service.ThriftServiceProcessor.<init>(ThriftServiceProcessor.java:76)
	at com.facebook.swift.service.ThriftServiceProcessor.<init>(ThriftServiceProcessor.java:65)
```

File: pom.xml (hunk 1 above)
Comment (srowen) [inline pom.xml]: Well, as I say, there's no way this PR can be merged like this anyway, so please close it?

Comment (microhello) [inline pom.xml]: OK

**Comment by andrewor14** [comment]
@microhello please file a JIRA and add it to the title of this PR. See how other patches are opened.

**Comment by microhello** [comment]
@andrewor14  I have created a issue: https://issues.apache.org/jira/browse/SPARK-12548

**Comment by srowen** [comment]
Yeah the issue here is that this is already a "duplicate" of the issue to remove the Guava exceptions anyway

**Comment by andrewor14** [comment]
Also, I just noticed this is opened against the wrong branch. Please close this PR and re-open it against master if we do decide to continue work on this issue.

**Pull request closed without merging**